src/__tests__/EventuallyQueue-test.js

```diff
@@ -25,6 +25,9 @@ class MockObject {
   toJSON() {
     return this.attributes;
   }
+  _getSaveJSON() {
```

Comment:
What's the diff to `toJSON`?

Reply:
The main thing that makes this work is that `_getSaveJSON()` (outside the mock) does not include the `ACL` object while `toJSON` does. In the mocks, there is no real different why the tests didn't fail already they don't 

When using saveEventually and connection is down, the object gets stored in the queue as json. Today it uses 'getJSON' and that includes a json representation of ACL. `_getSaveJSON` doesn't. 

This json object is later used (line 276) to save the queued object when the server is back up:

`await object.save(queueObject.object, queueObject.serverOptions);`

save here does not support ACL in json format (or ACL at all, don't think you can change access outside of server). 

There are TBH probably even better fixes here. We should probably only store the dirty fields of the object and not overwrite everything (but it does already make sure before line 276 that its not overwriting other things).

Reply:
Reason why the EventuallyQueue tests didn't catch this is because ACLs aren't on the objects before they are loaded from the servers.

Doing this..

```
x = new Parse.Object('TestObject');
  ParseObject2 {id: undefined, _localId: undefined, _objCount: 1762, className: 'TestObject'}
x.set('foo', 'bar');
x.toJSON()
{foo: 'bar'}
x.save();
```

Would work fine. But in reality you often modify existing objects. When objects are loaded from server, the `toJSON` includes a lot more properties. Here is the same object:

```
x = await new Parse.Query('TestObject').find();
x[0].toJSON()

   {"foo":"bar",
    "createdAt":"2024-03-17T20:52:35.213Z",
    "updatedAt":"2024-03-17T20:53:30.133Z",
    "ACL":{"*":{"read":true,"write":true}},
    "objectId":"sZhK7r5owf"}

```

All tests in `EventuallyQueue-test` just creates non-server saved objects which doesn't catch this problem with "real" objects from servers.

example:
```
const object = new ParseObject('TestObject');
    object.set('foo', 'bar');
    object.set('test', '1234');
```